### ISSUE TITLE
My attempts to read information from Blazored.LocalStorage fail.

### ISSUE THREAD

**Issue by MetodiEterov** [Blazored/LocalStorage#181]
**Describe the bug**
Hello,

I'm trying to use your solution for managing information in browser localstorage, but when I writing large amounts of information (about 35k!) to a variable in the localstorage, often my attempts to read it just stuck.
This stuckking goes on indefinitely.
I have followed your instructions when implementing your solution.
No errors are generated for me. The application just stops.
I can see and read the variable in the browser through the console, but not with your solution.

**To Reproduce**

Added in Startup.cs file:
services.AddBlazoredLocalStorage();

Injected the service in a Blazor component:
[Inject]
protected Blazored.LocalStorage.ILocalStorageService _localStorage { get; set; }

And called the service in OnAfterRenderAsync method:
protected override async Task OnAfterRenderAsync(bool firstRender)
{
    if (firstRender)
    {
        var messages = await _localStorage.GetItemAsStringAsync("ServerMessages");
    }
}

No errors!!!
The application stuck!
The only error in a browser console is: "Error: Connection disconnected with error 'Error: Server returned an error on close: Connection closed with an error.'."(blazor.server.js)

**Expected behavior**
I expect something simple: if errors occur, whatever their source, I need to be informed about them, and not somewhere something happens and I have to guess what it is...

**Screenshots**

**Hosting Model (is this issue happening with a certain hosting model?):**
My application is based on Blazor Server, .Net 6.0 framework.
I installed Blazored.LocalStorage Version="4.2.0".

**Comment by giovannigambino1980** [comment]
Hi, I'm having the same issue.
Even if the row is in a try/catch block, I don't get any exception, just a disconnection from the server.
Any Idea?

**Comment by chrissainty** [comment]
If one of you can provide a repro of the issue I can take a look.

**Comment by OrangeGoblin1973** [comment]
Same here. It is setting on the first run, then after that, it just freezes, no exceptions, it just disconnects. The application I have runs on a timer for 60 seconds, then refreshes the page.

As the first run means the cache is empty, it set's it fine. When the page is called again, it will not pass `await _localStorage.GetItemAsync<DateTime?>(SomeModelDateCache);` and just hangs there, and appears to be disconnected. 

Unfortunately, I can't add a repo. 

I am using Firefox: 104.0 (64-bit)
VS22: Version 17.3.2
Blazored Storage: 4.2.0

************************************

    '[Inject] public ILocalStorageService _localStorage { get; set; }

    protected override async Task OnAfterRenderAsync(bool firstRender)
    {
	   if (firstRender)
	   {
		if (!string.IsNullOrEmpty(DeviceName))
		{
			try
			{
				//This is where it stops on the next run. 
				SomeModel = await _localStorage.GetItemAsync<SomeModel>("SomeModelCache");
				DateTime? cacheDate = await _localStorage.GetItemAsync<DateTime?>(SomeModelDateCache);
				double totalTime = 0;
				
				if (cacheDate is not null)
				{
					totalTime = DateTime.UtcNow.Subtract((DateTime)cacheDate).TotalHours;
				}

				if (SomeModel is null || cacheDate is null || totalTime > 1)
				{
					await _localStorage.SetItemAsync<SomeModel>(SomeModelCache, someModelData);
					await _localStorage.SetItemAsync(SomeModelDateCache, DateTime.UtcNow);
				}
			}
			catch (Exception ex)
			{
				var stop = 1;
			}
		}
       }
    }'

************************************

This is the config settings in Program.cs
     `
       builder.Services.AddBlazoredLocalStorage(config =>
      {
            config.JsonSerializerOptions.IgnoreReadOnlyProperties = true;
            config.JsonSerializerOptions.ReadCommentHandling = JsonCommentHandling.Skip;
            config.JsonSerializerOptions.WriteIndented = true;
      });`

**Comment by NUlliiON** [comment]
Same issue when storage contains 35kb of json :(

**Comment by giovannigambino1980** [comment]
Seems more than 50 elements in the array

Ottenere Outlook per Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by JonathanShrek** [comment]
Any update on this? I'm also having this issue. I have an object in local storage that contains a few lists and dictionaries and my server is just hanging with no real error. If I remove one of the dictionaries it works again. It seems to be related to the complexity of the object.

**Comment by mappiltd** [comment]
I managed to solve my problem. 

When I create the model and pass it to the service to be stored, the JSON was malformed. I put the JSON from the browser local storage through one of the online validators and it wasn't valid.

Maybe check the JSON in the browser storage. 

However, this didn't fully solve it.

I moved everything into a scoped service IStorage which injects ILocalStorage, the cache name is being set in the service constructor using IOptions.

I then inject IStorage into the razor component.

This solved it for me.

**Comment by JonathanShrek** [comment]
https://github.com/JonathanShrek/BlazoredLocalStorageErrorExample

I've created a barebones repo to reproduce the issue. It seems to be occurring with complex objects.

**Comment by chrissainty** [comment]
Thanks for providing a repro @JonathanShrek. I'm now about to go on holiday but I'll take a look at this once I'm back 👍

**Comment by mappiltd** [comment]
I think I have fixed it. 

You can either do this:
`     public async Task GetLocalStorage()
      {
            SessionID = await GetSessionID();
            //await LocalStorage.GetItemAsync<StorageModel>($"{SessionID}");
            await LocalStorage.GetItemAsync<StorageModel>($"\"{SessionID}\"");
        }`

or you can do this:
`    public async Task SetLocalStorage()
     {
            await LocalStorage.SetItemAsync($"\"{SessionID}\"", await BuildLocalStorageObject());
      }`

When I validated the JSON the session Id is a messy string with out interpolation, I am assuming it down to how it reads the name of the local storage cache.

**Comment by JonathanShrek** [comment]
Further digging revealed that the error was occurring at the "localStorage.getItem" .NET function call. I opened up a ticket with them and they responded referencing an old closed ticket here. My error was resolved by increasing the SignalR message size limit. I'll link my ticket I logged with asp.net core in case it helps anyone else having the same issue.

https://github.com/dotnet/aspnetcore/issues/43974
